hazelcast/src/main/java/com/hazelcast/client/cache/impl/OneShotExecutionCallback.java

```diff
@@ -1,97 +0,0 @@
-/*
- * Copyright (c) 2008-2019, Hazelcast, Inc. All Rights Reserved.
- *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- * http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-
-package com.hazelcast.client.cache.impl;
-
-import com.hazelcast.core.ExecutionCallback;
-import com.hazelcast.internal.util.Clock;
-
-import java.util.concurrent.TimeoutException;
-import java.util.concurrent.atomic.AtomicIntegerFieldUpdater;
-
-import static com.hazelcast.internal.util.ExceptionUtil.sneakyThrow;
-
-/**
- * A specific {@link ExecutionCallback} base implementation which has this behaviour:
- * <ul>
- * <li>If it has not been called yet, it will be called and it will be waited to finish.</li>
- * <li>If it has been called but not finished yet, it will be waited to finish.</li>
- * <li>If it has been called and finished already, it will return immediately.</li>
- * </ul>
- *
- * @param <V> type of the response
- */
-abstract class OneShotExecutionCallback<V> implements ExecutionCallback<V> {
```

Comment:
I remember that this was introduced to make sure a specific behavior on near cache. Did we find an alternative solution for that? 
See details and related test here https://github.com/hazelcast/hazelcast/pull/7167

Reply:
@sancar the behaviour has been restored in https://github.com/hazelcast/hazelcast/pull/15441/commits/074a439b9b03464ac23fcd7e3d88214783b7795a. So the `PutAsyncOneShotCallback` is executed only once, either as a registered callback or because a user invoked one of the blocking methods to obtain the completion value.